src/peft/tuners/lora/conversion.py

```diff
@@ -0,0 +1,348 @@
+
+    if num_modules_with_support == 0:
+        raise TypeError("Could not detect any layer that supports LoRA conversion.")
+
+    peft_config = getattr(model, "peft_config", {}).get(adapter_name)
+    if (peft_config is not None) and (peft_config.peft_type == PeftType.LORA):
+        warnings.warn(
+            "Converting a PEFT adapter to LoRA that is already a LoRA adapter. There is typically no need for that."
+        )
+
+    config_bias = getattr(peft_config, "bias", getattr(peft_config, "lora_bias", "none"))
+    if config_bias != "none":
+        # TODO: remove if/when we remove support for bias completely
+        raise ValueError(f"The adapter's config sets bias={config_bias}, this is not supported right now.")
+
+    ###############
+    # PREPARATION #
+    ###############
+
+    peft_prefix = "base_model.model."
+    config_kwargs = {
+        "rank_pattern": {},
+        "alpha_pattern": {},
+        "exclude_modules": set(),
+    }
+
+    if peft_config is not None:
+        # use the model's PEFT config, if it exists, to initialize the new LoraConfig
+        peft_config = model.peft_config[adapter_name]
+        config_kwargs["target_modules"] = copy.copy(peft_config.target_modules)
+        config_kwargs["base_model_name_or_path"] = peft_config.base_model_name_or_path
+        if hasattr(peft_config, "layers_pattern"):
+            # those two go hand in hand
+            config_kwargs["layers_pattern"] = peft_config.layers_pattern
+            config_kwargs["layers_to_transform"] = peft_config.layers_to_transform
+        if isinstance(rank, int):
+            # hard-coded rank
+            lora_config = LoraConfig(r=rank, lora_alpha=rank, **config_kwargs)
+        else:
+            # r and lora_alpha shouldn't matter, as the rank will be determined by rank/alpha pattern
+            lora_config = LoraConfig(r=1, lora_alpha=1, **config_kwargs)
+    else:
+        # create a new LoraConfig from scratch, inferring the target modules from the model
+        lora_config = LoraConfig(
+            r=rank if isinstance(rank, int) else 1,  # 1 is a dummy value, actual values will come from rank_pattern
+            target_modules=[],
+            **config_kwargs,
+        )
+
+    ##############
+    # CONVERSION #
+    ##############
+
+    state_dict = {}
+    for name, module in tqdm(
+        model.named_modules(), disable=not progressbar, desc="Converting to LoRA", total=num_modules_total
+    ):
+        if not isinstance(module, BaseTunerLayer):
+            continue
+
+        lora_A, lora_B, effective_rank = _convert_module_to_lora(module, rank=rank, adapter_name=adapter_name)
+        if effective_rank == 0:
+            # This shouldn't really happen, as we ensure that the rank is greater than 0 (int) or, for tresholds
+            # (float), at least one SV is included. But better be safe than sorry, as, in principle, it is fine to
+            # exclude some layers.
```

Comment:
Perhaps we log with a `logging.debug()`?

Reply:
We don't have any logging like this in PEFT anywhere. I think introducing it just for this is overkill.